libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -195,65 +189,51 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
     }
 
     private fun sessionStart() {
-        var freeDrive = false
+        var isActiveGuidance = true
         ifNonNull(weakMapboxNavigation.get()?.getRoutes()) { routes ->
-            freeDrive = routes.isEmpty()
+            isActiveGuidance = routes.isNotEmpty()
         }
-        when (freeDrive) {
+        when (isActiveGuidance) {
             true -> {
-                // Do nothing
-                Log.d(TAG, "Free drive not supported")
-            }
-            false -> {
                 telemetryThreadControl.scope.launch {
                     callbackDispatcher.resetRouteProgressProcessor()
                     postUserEventDelegate =
                         postUserFeedbackEventAfterInit // Telemetry is initialized and the user selected a route. Allow user feedback events to be posted
                     handleSessionStart()
-                    sessionEndPredicate = { sessionStop() }
+                    sessionEndPredicate = {
+                        telemetryScope.launch {
+                            sessionStop()
+                        }
+                    }
                 }
             }
+            false -> {
+                // Do nothing
+                Log.d(TAG, "Only Active Guidance supported")
+            }
         }
     }
 
-    private fun sessionStop() {
-        Log.d(TAG, "TripSessionState.STOPPED")
+    private suspend fun sessionStop() {
+        Log.d(TAG, "sessionStop")
         postUserEventDelegate =
             postUserEventBeforeInit // The navigation session is over, disallow posting user feedback events
```

Comment:
`postUserEventBeforeInit` looks worthless. Log to the console? Was this being called?

It's a debug log: "Not in a navigation session, cannot send user feedback events". Looks like an error

Reply:
If that's all it does.. 🎁  if you delete it

Reply:
It's a way to "reset" 👀 

```kotlin
/**
 * The lambda that is called if the SDK client did not initialize telemetry. If telemetry is not initialized,
 * calls to post a user feedback event will fail silently
 */
```

Reply:
yeah i only see a log didn't seem needed 🤷 
```
        { _, _, _, _, _ ->
            Log.d(TAG, "Not in a navigation session, cannot send user feedback events")
        }
```

Reply:
We can remove the log but it's "failing" silently or better said doing nothing if we tried to send events in an invalid state.

Reply:
Yeah up to you :). I can't tell if it's failing silently or successfully making noise 🙃